test/v2/load/load_generator.go

```diff
@@ -257,9 +261,29 @@ func (l *LoadGenerator) disperseBlob(rand *random.TestRandom) (
 		return corev2.BlobKey{}, nil, nil, fmt.Errorf("failed to compute blob key: %w", err)
 	}
 
+	// Estimate gas for CheckDACert call
+	go l.estimateGasCheckDACertAsync(eigenDAV3Cert)
+
 	return blobKey, payload, eigenDACert, nil
 }
 
+// estimateGasCheckDACertAsync performs gas estimation in a separate goroutine to avoid blocking blob dispersal.
+func (l *LoadGenerator) estimateGasCheckDACertAsync(eigenDAV3Cert *coretypes.EigenDACertV3) {
```

Comment:
```suggestion
// estimateAndReportGasCheckDACert performs gas estimation and reports it as a metric.
// Make sure to call this in a separate goroutine to avoid blocking blob dispersal.
func (l *LoadGenerator) estimateAndReportGasCheckDACert(eigenDAV3Cert *coretypes.EigenDACertV3) {
```

Reply:
the function is not async right..? it is blocking the way written, so needs to be called in goroutine

Reply:
suggested change included in [85e3469](https://github.com/Layr-Labs/eigenda/pull/1890/commits/85e3469d534a655ed598270c9e9f43fcbcafae39)